xonsh/procs/proxies.py

```diff
@@ -500,10 +500,16 @@ def run(self):
             sp_stderr = sys.stderr
         # run the function itself
         try:
+            alias_stack = builtins.__xonsh__.env.get("ALIAS_STACK", "")
+            if self.env["ALIAS_NAME"]:
+                alias_stack += ":" + self.env["ALIAS_NAME"]
+
             with STDOUT_DISPATCHER.register(sp_stdout), STDERR_DISPATCHER.register(
                 sp_stderr
             ), xt.redirect_stdout(STDOUT_DISPATCHER), xt.redirect_stderr(
                 STDERR_DISPATCHER
+            ), builtins.__xonsh__.env.swap(
+                ALIAS_STACK=alias_stack
             ):
```

Comment:
can this be more static? instead of relying on env variable as suggested by gforsyth.

Reply:
The short answer is no because we have ExecAlias. As you can see we use the env variable temporarily only during the executing the command.